rc/extra/ranger.kak

```diff
@@ -14,30 +14,30 @@ define-command \
   -docstring %{ranger [<arguments>]: open the file system explorer to select buffers to open
 All the optional arguments are forwarded to the ranger utility} \
   ranger %{ %sh{
-  if [ -n "$TMUX" ]; then
+  if [ -n "${TMUX}" ]; then
     tmux split-window -h \
-      ranger $@ --cmd " \
+      ranger "$@" --cmd " \
         map <return> eval \
           fm.execute_console('shell \
-            echo evaluate-commands -client $kak_client edit {file} | \
-            kak -p $kak_session; \
+            printf %s evaluate-commands -client ' + shell_escape('$kak_client') + ' edit {file} | \
+            kak -p ' + shell_escape('$kak_session') + '; \
             tmux select-pane -t $kak_client_env_TMUX_PANE'.format(file=fm.thisfile.path)) \
           if fm.thisfile.is_file else fm.execute_console('move right=1')"
-  elif [ -n "$kak_client_env_STY" ]; then
+  elif [ -n "${STY}" ]; then
     tty="$(ps -o tty ${kak_client_pid} | tail -n 1)"
     args="${@:-$PWD}"
     screen -X eval \
         'split -h' \
         'focus down' \
-        "screen ranger $args
+        "screen ranger \"$args\"
         --cmd '
 map <return> eval \
 fm.execute_console(\"shell \
-echo evaluate-commands -client ${kak_client} edit {file} | \
-kak -p ${kak_session};\
+printf %s evaluate-commands -client \" + shell_escape(\"${kak_client}\") + \" edit {file} | \
```

Comment:
Not sure how this works, as in theory everything after `%s` is supposed to be a single argument.

Reply:
> @lenormf: in theory everything after %s is supposed to be a single argument.

Bizarrely, infuriatingly, this stopped working consistently "on my machine" and now sometimes only produces a single concatenated argument.

`printf "%s "` or `printf %s$' ' [...]` or `printf %s "[...]"` produces the expected result consistently.

Reply:
I've reverted to using echo in the tmux and x11 sections so as not to break what was working in master.